buildbot-buildmaster/buildbot/master.cfg

```diff
@@ -0,0 +1,276 @@
+        'debug_websockets' : 'BUILDBOT_MQ_DEBUG' in os.environ,
+        'debug_lowlevel' : 'BUILDBOT_MQ_DEBUG' in os.environ,
+    }
+# 'protocols' contains information about protocols which master will use for
+# communicating with workers. You must define at least 'port' option that workers
+# could connect to your master with this protocol.
+# 'port' must match the value configured into the workers (with their
+# --master option)
+c['protocols'] = {'pb': {'port': os.environ.get("BUILDBOT_WORKER_PORT", 9989)}}
+
+####### CHANGESOURCES
+
+# the 'change_source' setting tells the buildmaster how it should find out
+# about source code changes.  Here we point to the buildbot clone of pyflakes.
+
+testRepoName = "data-compressor-tests"
+testRepoOwner = 'CenterForSecureEnergyInformatics'
+testRepo = "https://github.com/" + testRepoOwner + "/" + testRepoName + ".git"
+
+compressorRepoName = 'data-compressor'
+compressorRepoOwner = 'CenterForSecureEnergyInformatics'
+compressorRepo = "https://github.com/" + compressorRepoOwner + "/" + compressorRepoName + ".git"
+
+#compressorRepo = 'https://github.com/CenterForSecureEnergyInformatics/data-compressor.git'
+token = open(secretsdir + "/githubToken").read().strip()
+
+c['change_source'] = []
+c['change_source'].append(changes.GitPoller(
+        compressorRepo,
+        workdir='gitpoller-workdir',
+        branches=['master'],
+        pollInterval=30))
+c['change_source'].append(changes.GitHubPullrequestPoller(
+	owner=compressorRepoOwner,
+	repo=compressorRepoName,
+	pollInterval=30,
+	token=token))
+
+####### SCHEDULERS
+
+# Configure the Schedulers, which decide how to react to incoming changes.  In this
+# case, just kick off a 'runtests' build
+
+c['schedulers'] = []
+c['schedulers'].append(schedulers.AnyBranchScheduler(
+                            name="anyBranch",
+                            treeStableTimer=None,
+                            builderNames=["x86_64","i386","armv6_native","armv7_native","armv8_native","arm64v8_native","arm","win_x64","win32"]))
+c['schedulers'].append(schedulers.ForceScheduler(
+                            name="force",
+                            builderNames=["x86_64","i386","armv6_native","armv7_native","armv8_native","arm64v8_native","arm","win_x64","win32"]))
+
+####### FACTORIES
+testWorkdirName = testRepoName
+testdir = "../" + testWorkdirName + "/tests/"
+
+desc = "building and testing"
+descDone = "build and test"
+
+windowsShell = "C:\\Program Files\\Git\\bin\\bash.exe"
+
+factoryLinux_x86_64 = util.BuildFactory()
+factoryLinux_x86_64.addStep(steps.Git(name="git - tests", repourl=testRepo, workdir=testWorkdirName, mode='full', alwaysUseLatest='true'))
+factoryLinux_x86_64.addStep(steps.Git(name="git - data-compressor", repourl=compressorRepo, workdir='data-compressor', mode='full', alwaysUseLatest='true'))
+factoryLinux_x86_64.addStep(steps.ShellCommand(description=desc, descriptionDone=descDone, command=[testdir + "checkBits.sh", "x86_64"]))
```

Comment:
Can the following commands be simplified? There seems to be a lot of duplicated code.

Reply:
Reduced the Code a little. The way steps are added now reflects the way shown in the buildbot manual: http://docs.buildbot.net/latest/manual/configuration/steps/index.html#